service/src/main/java/bio/terra/workspace/service/resource/controlled/GcsApiConversions.java

```diff
@@ -208,12 +208,18 @@ public static ApiGcpGcsBucketLifecycleRuleActionType toWsmApi(String lifecycleAc
   public static LifecycleCondition toGcsApi(ApiGcpGcsBucketLifecycleRuleCondition condition) {
     final var resultBuilder = LifecycleCondition.newBuilder();
 
-    /* TODO(PF-506): some conditions aren't in the version of the Google Storage API in the
-     *    latest version of the CRL. */
     resultBuilder.setAge(condition.getAge());
     resultBuilder.setCreatedBefore(toGoogleDateTime(condition.getCreatedBefore()));
     resultBuilder.setNumberOfNewerVersions(condition.getNumNewerVersions());
     resultBuilder.setIsLive(condition.isLive());
+    resultBuilder.setMatchesStorageClass(
+        Optional.ofNullable(condition.getMatchesStorageClass())
+            .map(c -> c.stream().map(GcsApiConversions::toGcsApi).collect(Collectors.toList()))
+            .orElse(null));
```

Comment:
It looks like there's a duplicate of this below. Can you remove one?

Reply:
good catch, thanks!